### ISSUE TITLE
[Dynamic Form] Update Form crashes on loading,  if an item has TaxonomyFieldTypeMulti field.  [3.16.1] !!! 

### ISSUE THREAD

**Issue by bva999** [pnp/sp-dev-fx-controls-react#1733]
#### Category
[ ] Enhancement

[x ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [3.16.1]

#### Observed Behavior
If an item has a TaxonomyFieldTypeMulti  field , the Dynamic Form crashes on loading. And get next error: 

```
`Error get field informations TypeError: defaultValue.split is not a function
    at eval (DynamicForm.js:591:1)
    at step (DynamicForm.js:58:1)
    at Object.eval [as next] (DynamicForm.js:39:45)
    at step (DynamicForm.js:43:1)
    at Object.eval [as next] (DynamicForm.js:39:45)
    at fulfilled (DynamicForm.js:30:42)`
```
It assumes that value should be a string , but it is an Array of objects: 

![Screenshot 2024-01-09 135139](https://github.com/pnp/sp-dev-fx-controls-react/assets/12149939/06daeb3d-f663-488c-bca9-292bc81894c2)

**Comment by michaelmaillot** [comment]
Hi @bva999,

I tried to repro without success but maybe I'm missing context info.

Can you give me details about the Taxonomy field you're refering to, such as column settings, term structure (parent / children),...?

In my case I tried with following config (sorry it's in french):

![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/44271458/d9168195-0b76-421e-a1f6-9e2998e04553)

![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/44271458/2eb69783-acf4-41b6-9eb4-d09d3dba7a0e)

![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/44271458/1d248c7b-b8ad-4880-9869-172ffc6b30b0)

**Comment by michaelmaillot** [comment]
Ok I can repro the issue.

If I'm not mistaken, we shouldn't end up in the last __else__ block (in the screen) as this one is reached for a new item. We should be inside the first __if__ block that parses the Taxonomy array values. And neither you and me have a property called "results" in the `defaultValue`.

@IRRDC following what you've stated [here](https://github.com/pnp/sp-dev-fx-controls-react/issues/1568#issuecomment-1663281413), I would like to get more info regarding your use case (list / library, field properties, site context,...), if you don't mind.

I want to understand what are we missing here, even if I realized that I didn't push far enough tests when merging the PR #1718...

**Comment by IRRDC** [comment]
@michaelmaillot Thanks for looking into this. I had to search for a while and noticed that back in August I got my information from my web part stopping on the error in the compiled code.
Here is the screenshot from back then:
![258001797-aa0fd8c4-592f-4f5d-af9e-315a4f850b8f](https://github.com/pnp/sp-dev-fx-controls-react/assets/11270504/f1fd886d-9c71-4a80-a390-4e259a1ea281)
While I'm directly debugging in the checked out TypeScript code where the data seems not to be split into __metadata and results:
![Zwischenablage01](https://github.com/pnp/sp-dev-fx-controls-react/assets/11270504/0e12ecfb-1e54-4f1b-ae08-3883767ee9c0)
So I guess a better check to see if item and item[field.InternalName] exist should have been enough.

**Comment by IRRDC** [comment]
I created a new fix (to fix the fix): https://github.com/pnp/sp-dev-fx-controls-react/pull/1736

**Comment by michaelmaillot** [comment]
Hi @IRRDC & @bva999,

A new version has been [released](https://www.npmjs.com/package/@pnp/spfx-controls-react/v/3.16.2) which includes this fix.

Can you check and tell me if that's ok for you?

**Comment by IRRDC** [comment]
@michaelmaillot  No, it doesn't. I'm back to my original problem, where in the compiled code item[field.InternalName] contains a sub-property "results" which has to be iterated. My original fix worked for that in the compiled code from the npm package but doesn't work while running the solution directly from the debugger. I guess to make it work in both environments we have to add two cases again that check the existence of "results".

**Comment by IRRDC** [comment]
Or even better: Attack the root of the problem and find out why the data is structured differently. This might cause problems in other controls, too, that haven't been noticed yet.

**Comment by IRRDC** [comment]
When you compare my two screenshots you can see that it also happens to MLookupLinksId, which - as the name suggests - is a multi lookup column.

**Comment by IRRDC** [comment]
Hm, I guess we can't hope for an update of the 3 year old version of pnp/sp that controls-react uses. Are there plans to update the solution to 3.22.0?

**Comment by michaelmaillot** [comment]
Thanks for this feedback. This is strange indeed and I couldn't reproduce the `.results` thing you're facing.

Are you suggesting that @pnp/sp 3.x returns data in a different format than 2.x version?

**Comment by IRRDC** [comment]
That's what I thought first but I haven't been able to reproduce the error in the newly created web part and have by now copied nearly all the code and all the settings from the one that throws the error (and produces the additional "results" property).

**Comment by michaelmaillot** [comment]
@IRRDC did you figure out the `results` origin?

**Comment by IRRDC** [comment]
No, it's very frustrating. I've spent days on it and created new working solutions and then suddenly the dynamic form failed to load again because the data was in "results" again. I thought I had a issue free version a few hours ago but it fails when testing on a different SharePoint tenent. I will create a new solution again tomorrow and see if it fails from the start on that tenent.

**Comment by IRRDC** [comment]
I started again with a new solution and a HelloWorld web part with just the dynamic form and it can display list A from site X on tenant 1 and list B on site Y on tenant 1 and list C on tenant 2.
Yesterday's solutions that started the same way and got the code of our actual web parts added to them can display a sub set of those three test lists. Some can display none of the three, some can only display list A, etc. It is consistent which solution can display which list. When it doesn't work it's because "results" got added to the data.
The only explanation I can come up with is that different versions of the very long list of packages don't get resolved perfectly when the code gets comiled into a single js file. I had a similar problem about a year ago when it was very important in which order I installed the spfx-controls-react and spfx-property-controls because they used different versions of pnp/sp and even though those different versions ended up in their node_modules folders one of them used not their own version but the one from the base folder. That caused errors during compilation which where much easier to spot.
Can we add a fallback for this weird case, like:
`if (item[field.internalname].hasownproperty('results')) { item[field.internalname] = item[field.internalname]['results']; }`
?

**Comment by IRRDC** [comment]
I created another fix and copied the compiled code into my solution where it is now able to load the form for all sites and lists. https://github.com/pnp/sp-dev-fx-controls-react/pull/1744

**Comment by IRRDC** [comment]
Any thoughts?

**Comment by IRRDC** [comment]
@joelfmrodrigues Why did you close the bug without comment when there is an open pull request to fix existing issues?

**Comment by joelfmrodrigues** [comment]
@IRRDC thanks for the heads up, I may have mistaken this with another issue from a different PR from @michaelmaillot 
Please give us a moment to check and I will reopen it if required.

**Comment by IRRDC** [comment]
During a quick test this morning I couldn't reproduce the issue with the new version but I could not perform all test scenarios due to this new issue in the dynamic form https://github.com/pnp/sp-dev-fx-controls-react/issues/1758

**Comment by michaelmaillot** [comment]
> @joelfmrodrigues Why did you close the bug without comment when there is an open pull request to fix existing issues?

Hi @IRRDC this issue has been closed because its original PR (which was a rollback from has been released in 3.16.1) has been merged and closed, then released in minor version.

Your PR is linked to a global behavior for which we didn't figure out the root cause, thus not linked to this issue in particular.

Before addressing this one, we need to understand what causes this `results` sub-array to appear unconsistently, not only providing a solution that can handle both scenarios. We'll keep your PR open meanwhile of course.

I'll open a new issue.

**Comment by michaelmaillot** [comment]
Done #1759
